analysis/specimen_diff_vqgan_new.py

```diff
@@ -0,0 +1,258 @@
+
+# import copy
+import os
+import pickle
+from analysis.replace_codes import get_entropy_ordering
+import numpy as np
+from taming.analysis_utils import Embedding_Code_converter
+# import os
+# from taming.data.phylogeny import Phylogeny
+from taming.loading_utils import load_config, load_phylovqvae, load_CWVQGAN
+from taming.data.custom import CustomTest as CustomDataset
+# from taming.models.cond_transformer import Phylo_Net2NetTransformer
+
+import taming.constants as CONSTANTS
+from taming.models.vqgan import VQModel
+from taming.models.cwautoencoder import CWmodelVQGAN
+from taming.modules.losses.lpips import LPIPS
+# from taming.modules.losses.phyloloss import parse_phyloDistances
+from taming.plotting_utils import get_fig_pth
+
+
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import ImageGrid
+
+import torch
+
+from omegaconf import OmegaConf
+import argparse
+import tqdm
+from pathlib import Path
+
+
+
+class KeyImageHelper:
+    def __init__(self, total_num_codes, keyimagerate):
+        self.total_num_codes = total_num_codes
+        self.keyimagerate = keyimagerate
+        
+        self.update = True
+    
+    
+    
+    def new_update(self, i):
+        if i%self.keyimagerate == 0:
+            self.update = True
+        
+        return self.update
+        
+    def isKeyImage(self, i):
+        if self.update:
+            self.update = False
+            return True
+            
+        return False
+    
+    
+            
+def populate_resnet_scores(resnet_model, dec_image, change):
+    if resnet_model is not None:
+        change["class"] = torch.argmax(resnet_model(dec_image), dim=1).item()
+    return change 
+
+
+
+@torch.no_grad()
+def main(configs_yaml):
+    yaml_path = configs_yaml.yaml_path
+    ckpt_path = configs_yaml.ckpt_path
+    DEVICE= configs_yaml.DEVICE
+    image_index1_= configs_yaml.image_index1
+    image_index2_= configs_yaml.image_index2
+    count= configs_yaml.count
+    # phyloDistances_string = configs_yaml.phyloDistances_string
+    # phylogeny_path = configs_yaml.phylogeny_path
+    size= configs_yaml.size
+    file_list_path= configs_yaml.file_list_path
+    keyimgrate= configs_yaml.keyimgrate
+    # plt_path = configs_yaml.plt_path
+    
+    bb_model_path = configs_yaml.bb_model_path
+        
+    plt_path = "default"
+    
+    # phylo_distances = parse_phyloDistances(phyloDistances_string)
+    
+    # load image
+    dataset = CustomDataset(size, file_list_path, add_labels=True)
```

Comment:
Please rename this file to something like /home/elhamod/projects/phylonn/analysis/specimen_translation.py and /home/elhamod/projects/phylonn/analysis/specimen_translation_vqgan.py

Reply:
same goes for the yaml file

Reply:
adding new argument for that

Reply:
done